### ISSUE TITLE
[image_picker] Bunch of different errors 

### ISSUE THREAD

**Issue by mrigankdoshy** [flutter/flutter#74196]
My app was working just fine on the simulator, until I tried to run it on my iPhone. Then it started giving these errors and now it does not run on the simulator either. 

## Steps to Reproduce

This is specific to my app. There seems to be something wrong with the image_picker. I'm unsure of how to give steps on how to reproduce the issue.

**Expected results:**  I except it to run my app (it used to before)

**Actual results:**  
```
Running pod install...                                              3.1s
Running Xcode build...                                                  
Xcode build done.                                            9.2s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/image_picker-0.6.7+21/ios/Classes/FLTImagePickerPlugin.m:149:20: warning: 'UIAlertView' is deprecated:
    first deprecated in iOS 9.0 - UIAlertView is deprecated. Use UIAlertController with a preferredStyle of UIAlertControllerStyleAlert instead [-Wdeprecated-declarations]
        [[[UIAlertView alloc] initWithTitle:NSLocalizedString(@"Error", nil)
                       ^
    In module 'UIKit' imported from /Users/mrigankdoshy/Documents/Work/Business/Whispering Mule/App/mule-app/ios/Pods/Target Support
    Files/image_picker/image_picker-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIA
    lertView.h:27:12: note: 'UIAlertView' has been explicitly marked deprecated here
    @interface UIAlertView : UIView
               ^
    1 warning generated.
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/image_picker-0.6.7+21/ios/Classes/FLTImagePickerPhotoAssetUtil.m:17:44: warning:
    'UIImagePickerControllerReferenceURL' is deprecated: first deprecated in iOS 11.0 - Will be removed in a future release, use PHPicker. [-Wdeprecated-declarations]
      NSURL *referenceURL = [info objectForKey:UIImagePickerControllerReferenceURL];
                                               ^
    In module 'UIKit' imported from /Users/mrigankdoshy/Documents/Work/Business/Whispering Mule/App/mule-app/ios/Pods/Target Support
    Files/image_picker/image_picker-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UII
    magePickerController.h:62:51: note: 'UIImagePickerControllerReferenceURL' has been explicitly marked deprecated here
    UIKIT_EXTERN UIImagePickerControllerInfoKey const UIImagePickerControllerReferenceURL API_DEPRECATED("Will be removed in a future release, use PHPicker.", ios(4.1, 11.0))
    API_UNAVAILABLE(tvos); // an NSURL that references an asset in the AssetsLibrary framework
                                                      ^
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/image_picker-0.6.7+21/ios/Classes/FLTImagePickerPhotoAssetUtil.m:21:47: warning:
    'fetchAssetsWithALAssetURLs:options:' is deprecated: first deprecated in iOS 11 - Will be removed in a future release [-Wdeprecated-declarations]
      PHFetchResult<PHAsset *> *result = [PHAsset fetchAssetsWithALAssetURLs:@[ referenceURL ]
                                                  ^
    In module 'Photos' imported from /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/image_picker-0.6.7+21/ios/Classes/FLTImagePickerPhotoAssetUtil.h:6:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.2.sdk/System/Library/Frameworks/Photos.framework/Headers/PH
    Asset.h:74:1: note: 'fetchAssetsWithALAssetURLs:options:' has been explicitly marked deprecated here
    + (PHFetchResult<PHAsset *> *)fetchAssetsWithALAssetURLs:(NSArray<NSURL *> *)assetURLs options:(nullable PHFetchOptions *)options API_DEPRECATED("Will be removed in a
    future release", ios(8, 11), tvos(8, 11)) API_UNAVAILABLE(macos);
    ^
    2 warnings generated.
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/ios/Classes/PermissionManager.m:69:93: error: expected expression
            BOOL success = [[UIApplication sharedApplication] openURL:options:completionHandler::[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
                                                                                                ^
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/ios/Classes/PermissionManager.m:69:67: error: use of undeclared identifier
    'options'
            BOOL success = [[UIApplication sharedApplication] openURL:options:completionHandler::[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
                                                                      ^
    /Users/mrigankdoshy/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/ios/Classes/PermissionManager.m:69:75: error: use of undeclared identifier
    'completionHandler'
            BOOL success = [[UIApplication sharedApplication] openURL:options:completionHandler::[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
                                                                              ^
    3 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

```

<details>
  <summary>Logs</summary>

 Output to large for terminal to show everything

```
```

```
warning • Invalid option specified for 'enable-experiment': Flag "spread-collections" is no longer required • analysis_options.yaml:3:5 • invalid_option
warning • Invalid option specified for 'enable-experiment': Flag "spread-collections" is no longer required • analysis_options.yaml:3:5 • invalid_option
   info • Unused import: 'dart:convert' • lib/config/http_client.dart:1:8 • unused_import
   info • Unused import: 'dart:typed_data' • lib/config/http_client.dart:3:8 • unused_import
   info • Unused import: 'package:mule/models/res/requestedFromMeRes/requested_from_me_res.dart' • lib/config/http_client.dart:19:8 • unused_import
   info • The declaration '_makeGetRequest' isn't referenced • lib/config/http_client.dart:36:20 • unused_element
   info • This function has a return type of 'FutureOr<ImageProvider<Object>>', but doesn't end with a return statement • lib/config/http_client.dart:218:25 • missing_return
   info • The value of the local variable 'number' isn't used • lib/config/http_client.dart:220:9 • unused_local_variable
   info • Duplicate import • lib/models/data/order_data.dart:3:8 • duplicate_import
   info • 'List' is deprecated and shouldn't be used. Use a list literal, [], or the List.filled constructor instead • lib/models/res/errorRes/error_res.dart:14:18 •
          deprecated_member_use
   info • Unused import: 'dart:io' • lib/screens/home/map/map_helper.dart:2:8 • unused_import
   info • The value of the field '_areMarkersLoading' isn't used • lib/screens/home/map/map_widget.dart:43:8 • unused_field
   info • The value of the local variable 'positionStream' isn't used • lib/screens/home/map/map_widget.dart:79:36 • unused_local_variable
   info • Unused import: 'package:flutter_progress_button/flutter_progress_button.dart' • lib/screens/home/slider/match/waiting_to_match_panel.dart:3:8 • unused_import
   info • The value of the local variable 'score' isn't used • lib/screens/home/slider/request/make_request_panel.dart:157:21 • unused_local_variable
   info • The value of the local variable 'currentLocation' isn't used • lib/screens/home/slider/request/make_request_panel.dart:160:30 • unused_local_variable
   info • Unused import: 'package:sliding_up_panel/sliding_up_panel.dart' • lib/screens/home/slider/search/search_panel.dart:12:8 • unused_import
   info • Unused import: 'dart:math' • lib/screens/home/slider/sliding_up_widget.dart:1:8 • unused_import
   info • Unused import: 'package:mule/config/app_theme.dart' • lib/screens/home/slider/sliding_up_widget.dart:4:8 • unused_import
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/ForgotPassword/email_otp.dart:162:30 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/ForgotPassword/forgot_password_email.dart:150:20 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/ForgotPassword/reset_password.dart:154:34 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/login_screen.dart:148:30 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/login_screen.dart:191:30 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/login/login_screen.dart:281:20 • deprecated_member_use
   info • Unused import: 'package:google_maps_flutter/google_maps_flutter.dart' • lib/screens/menu/home_drawer.dart:4:8 • unused_import
   info • This function has a return type of 'ImageProvider<Object>', but doesn't end with a return statement • lib/screens/menu/home_drawer.dart:459:17 • missing_return
   info • Unused import: 'package:get_it/get_it.dart' • lib/screens/orders/orders_screen.dart:2:8 • unused_import
   info • Unused import: 'package:mule/stores/global/user_info_store.dart' • lib/screens/orders/orders_screen.dart:7:8 • unused_import
   info • The declaration '_updateOrders' isn't referenced • lib/screens/orders/orders_screen.dart:28:3 • unused_element
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/profile/Account/change_email.dart:92:20 • deprecated_member_use
   info • Unused import: 'package:get_it/get_it.dart' • lib/screens/profile/Account/change_password.dart:3:8 • unused_import
   info • Unused import: 'package:mule/screens/profile/profile.dart' • lib/screens/profile/Account/change_password.dart:9:8 • unused_import
   info • Unused import: 'package:mule/stores/global/user_info_store.dart' • lib/screens/profile/Account/change_password.dart:10:8 • unused_import
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/profile/Account/change_password.dart:151:34 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/profile/Account/change_phone.dart:105:18 • deprecated_member_use
   info • The value of the local variable 'errRes' isn't used • lib/screens/profile/Account/delete_account.dart:38:16 • unused_local_variable
   info • Unused import: 'package:mule/screens/profile/Account/change_email.dart' • lib/screens/profile/profile.dart:5:8 • unused_import
   info • Unused import: 'package:mule/screens/profile/Account/change_phone.dart' • lib/screens/profile/profile.dart:7:8 • unused_import
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/signup/phone_otp.dart:123:28 • deprecated_member_use
   info • Unused import: 'package:mule/screens/signup/phone_otp.dart' • lib/screens/signup/signup_screen.dart:11:8 • unused_import
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/screens/signup/signup_screen.dart:169:24 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/widgets/alert_widget.dart:15:15 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/widgets/confirm_dialogue.dart:13:11 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/widgets/confirm_dialogue.dart:29:11 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/widgets/order_completion_dialogue.dart:13:11 • deprecated_member_use
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature
          was deprecated after v1.25.0-8.1.pre. • lib/widgets/order_completion_dialogue.dart:27:11 • deprecated_member_use
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: StylizedButton.callback •
          lib/widgets/stylized_button.dart:4:7 • must_be_immutable
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: StylizedButton.callback •
          lib/widgets/stylized_button.dart:43:7 • must_be_immutable
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: StylizedButton.callback •
          lib/widgets/stylized_button.dart:55:7 • must_be_immutable
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: StylizedButton.callback •
          lib/widgets/stylized_button.dart:67:7 • must_be_immutable
```

```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.401, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 1.26.0-2.0.pre.401 at /Users/mrigankdoshy/flutter
    • Framework revision da5a454d5c (34 hours ago), 2021-01-17 01:24:03 -0500
    • Engine revision 609036f2bf
    • Dart version 2.12.0 (build 2.12.0-236.0.dev)

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2020.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 47.1.3
    • Dart plugin version 201.8538.45

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • iPhone 11 Pro Max (mobile) • 173C06F5-D4CA-4450-AEAD-FCA64BB010CD • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-2 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 87.0.4280.88
```

</details>

**Comment by zenkog** [comment]
I got the same issue too. Have wasted 2 days on this. Can you tell me how did you fix it?
What I have done:
1. Upgrade all dependencies
2. Make sure all versions are latest one: Flutter (Stable channel), Android studio, SDKs, Xcode
3. in the Podfile I also have ensured the target version to be 10.12
4. Flutter clean
5. delete Pods and Podfile and Podfile lock and reinstall pods
6. Pod repo update

None of these worked.

**Comment by mrigankdoshy** [comment]
@zenkog I ultimately factory reseted my laptop (to reset my XCode settings) and then reinstalled flutter and everything worked smoothly. So I believe this has something to do with XCode settings. If your codebase is small, I would recommend that you create a project and copy over your code.

**Comment by zenkog** [comment]
@mrigankdoshy Thanks for the pointer. 
I have done what you said to Factory reset XCode:
1. Enter in terminal `defaults delete com.apple.Xcode`
2. Then delete the settings: `rm -rf ~/Library/Application\ Support/Xcode`

But it still doesn't work. 

Unless there is really no other choice, I want to refrain from factory reset my Mac (which will erase all the hard drive contents). The code base is medium size (about 346 items inside `lib` folder), but it's almost a year of work. 

Hope someone could help me out here. 

I'm not sure, but is this related to the MacOS update to Big Sur? Because all these just happened after that update.

**Comment by darshankawar** [comment]
@mrigankdoshy 
Are you still facing the issue ? Does it occur in debug or release mode ?
If so, can you provide full logs by running `flutter run --verbose` or `flutter run --release --verbose` ?

**Comment by zenkog** [comment]
@darshankawar I know you didn't ask me, but let me provide my `flutter run --verbose` in the file attached (too long and it prevent me to post it). Click the file below:

[flutter-error.pdf](https://github.com/flutter/flutter/files/5892487/flutter-error.pdf)

Thanks for looking into the problem

**Comment by mrigankdoshy** [comment]
@darshankawar Nope, I got rid of it. I think it occurred when Xcode settings were release mode (I'm not sure tho). My ```flutter run --verbose``` output is in the issue description

**Comment by darshankawar** [comment]
@zenkog 
Can you provide your `flutter doctor -v` ? the log you provided is indeed too long (817 pages) and practically impossible to go through each page to look at the error.  I see below which potentially is part of the error:

`Command PhaseScriptExecution failed with a nonzero exit` but this is very vague and doesn't reveal much.
Also, are you on Big Sur on which you are seeing this ? Is this only happening using `image_picker` ? If you remove it, does the issue occur ?

**Comment by zenkog** [comment]
@darshankawar Here is my `flutter doctor -v`:
```
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 1.22.6 at /Users/ZK/development/flutter
    • Framework revision 9b2d32b605 (7 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Volumes/ZK-R/Applications/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Volumes/ZK-R/Applications/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • ZK iPho (mobile) • b3ebbe93864b12b2d78acd5efecf25 • ios • iOS 14.3

• No issues found!
```

It is practically impossible for me to remove `image_picker` from my code. Because it is integrated in many parts of my code (it is a year worth of work). 

Yes I'm running on Big Sur. And it all went well exactly before I updated my Mac. 
I do not even recognize this `Command PhaseScriptExecution` I don't think I ever used it.

**Comment by zenkog** [comment]
I just tried to create a fresh new flutter app and run it, it runs well.
But when I added these in `pubspec.yaml`:
```
  flutter_form_builder: ^4.2.0
  image_picker: ^0.6.7+21

  # Firebase packages
  firebase_core: ^0.7.0
  firebase_auth: ^0.20.0+1
  cloud_firestore: ^0.16.0
  firebase_storage: ^7.0.0
  firebase_database: ^6.0.0
  cloud_functions: ^0.9.0
  firebase_admob: ^0.11.0+1
  firebase_analytics: ^7.0.1
```
And of course set the deployment target to 10.12 in the Podfile
There is no modification whatsoever in the `main.dart`
The app immediately closes. (blink white screen then close).

I'm doing more testing now.

**Comment by zenkog** [comment]
When I'm using only `image_picker` as dependency then it worked. So it is not from the `image_picker` itself.
I created another [issue here](https://github.com/flutter/flutter/issues/75045#issue-797321424) to follow up this problem

**Comment by darshankawar** [comment]
@zenkog 
Thanks for filing new issue. I am closing this one because the OP is not seeing the issue as their comment above, https://github.com/flutter/flutter/issues/74196#issuecomment-769957925.
